# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIROps.yaml
# Commit: 72bf9c871ded
# Full Hash: 72bf9c871ded006f56f09b24d0723365516445bb
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 5
# Description:
#   Bug 1692065 - wasm: Add tagging for 31-bit integers and i31ref. r=yury
#   
#   This commit adds a tag 0x1 for 31-bit integers in AnyRef. Any JS value
#   that is losslessly representable (without coercion) in this range is
#   stored without boxing. This is chosen to line up exactly with how the
# ==============================================================================

diff -r 28feda14fdec -r 72bf9c871ded js/src/jit/LIROps.yaml
--- a/js/src/jit/LIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit/LIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
@@ -3579,6 +3579,7 @@
   result_type: WordSized
   operands:
     input: BoxedValue
+  num_temps: 1
 
 - name: WasmAnyRefFromJSObject
   result_type: WordSized
@@ -3590,6 +3591,18 @@
   operands:
     input: WordSized
 
+- name: WasmNewI31Ref
+  mir_op: true
+  result_type: WordSized
+  operands:
+    value: WordSized
+
+- name: WasmI31RefGet
+  mir_op: true
+  result_type: WordSized
+  operands:
+    value: WordSized
+
 # Constant Simd128
 - name: Simd128
   result_type: WordSized